packages/bff/src/services/agreementService.ts

```diff
@@ -57,3 +146,240 @@ export const getLatestAgreement = async (
     }
   })[0];
 };
+
+function isUpgradable(
+  descriptor: catalogApi.EServiceDescriptor,
+  agreement: agreementApi.Agreement,
+  descriptors: catalogApi.EServiceDescriptor[]
+): boolean {
+  return descriptors
+    .filter((d) => parseInt(d.version, 10) > parseInt(descriptor.version, 10))
+    .some(
+      (d) =>
+        (d.state === "PUBLISHED" || d.state === "SUSPENDED") &&
+        (agreement.state === "ACTIVE" || agreement.state === "SUSPENDED")
+    );
+}
+
+async function enrichAgreement(
+  agreement: agreementApi.Agreement,
+  clients: PagoPAInteropBeClients,
+  ctx: WithLogger<BffAppContext>
+): Promise<bffApi.AgreementListEntry> {
+  const { consumer, producer, eservice } = await parallelGet(
+    agreement,
+    clients,
+    ctx
+  );
+
+  const currentDescriptior = getCurrentDescriptor(eservice, agreement);
+
+  return {
+    id: agreement.id,
+    state: agreement.state,
+    consumer: {
+      id: consumer.id,
+      name: consumer.name,
+      kind: consumer.kind,
+    },
+    eservice: getCompactEservice(eservice, producer),
+    descriptor: getCompactDescriptor(currentDescriptior),
+    canBeUpgraded: isUpgradable(
+      currentDescriptior,
+      agreement,
+      eservice.descriptors
+    ),
+    suspendedByConsumer: agreement.suspendedByConsumer,
+    suspendedByProducer: agreement.suspendedByProducer,
+    suspendedByPlatform: agreement.suspendedByPlatform,
+  };
+}
+
+export async function enhanceAgreement(
+  agreement: agreementApi.Agreement,
+  clients: PagoPAInteropBeClients,
+  ctx: WithLogger<BffAppContext>
+): Promise<bffApi.Agreement> {
+  const { consumer, producer, eservice } = await parallelGet(
+    agreement,
+    clients,
+    ctx
+  );
+
+  const currentDescriptior = getCurrentDescriptor(eservice, agreement);
+
+  const activeDescriptor = eservice.descriptors
+    .toSorted((a, b) => parseInt(a.version, 10) - parseInt(b.version, 10))
```

Comment:
In other parts of the app we do:
```suggestion
    .toSorted((a, b) => Number(a.version) - Number(b.version))
```